mcs/class/Mono.Posix/Mono.Unix.Native/Syscall.cs

```diff
@@ -1416,6 +1440,177 @@ public struct Linger {
 		public int l_linger;
 	}
 
+	[Map]
+	[StructLayout (LayoutKind.Sequential)]
+	[CLSCompliant (false)]
+	public struct InAddr : IEquatable<InAddr> {
+		public uint s_addr;
+
+		public InAddr (byte b0, byte b1, byte b2, byte b3)
+		{
+			s_addr = 0;
+			this[0] = b0;
+			this[1] = b1;
+			this[2] = b2;
+			this[3] = b3;
+		}
+
+		public unsafe InAddr (byte[] buffer)
+		{
+			if (buffer.Length != 4)
+				throw new ArgumentException ("buffer", "buffer.Length != 4");
+			s_addr = 0;
+			fixed (uint* ptr = &s_addr)
+				Marshal.Copy (buffer, 0, (IntPtr) ptr, 4);
+		}
+
+		public unsafe void CopyTo (byte[] destination, int startIndex)
```

Comment:
> You have a struct which is mutable (always dubious...), and a way to set the bytes (via the indexer), and a way to copy the bytes out...but no way to copy them in, other than the constructor.
> 
> I'm wondering...lots of things, but should there be a Copy(byte[], int) overload for copying int InAddr?
> 
> Alternatively, should this instance instead be immutable, and the indexer setter should be removed?

Most (or all?) of the other structs in Mono.Unix.Native are also mutable, so I'll add a CopyFrom(byte[], int) method.